packages/web3-eth/types/tests/eth.tests.ts

```diff
@@ -0,0 +1,414 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file eth-tests.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import {Eth, Log, Subscribe, BlockHeader, Transaction, Syncing, Block, TransactionReceipt, SignedTransaction} from 'web3-eth';
+
+const eth = new Eth('http://localhost:8545');
+
+// $ExpectType Contract
+eth.Contract;
+
+// $ExpectType Iban
+eth.Iban;
+
+// $ExpectType Personal
+eth.personal;
+
+// $ExpectType Accounts
+eth.accounts;
+
+// $ExpectType any
+eth.ens;
+
+// $ExpectType AbiCoder
+eth.abi;
+
+// $ExpectType Network
+eth.net;
+
+// $ExpectType boolean
+eth.clearSubscriptions;
+
+// $ExpectType Promise<Subscribe<Log>>
+eth.subscribe('logs');
+// $ExpectType Promise<Subscribe<Log>>
+eth.subscribe('logs');
+// $ExpectType Promise<Subscribe<Log>>
+eth.subscribe('logs', (error: Error, result: Subscribe<Log>) => {});
+// $ExpectType Promise<Subscribe<Log>>
+eth.subscribe('logs', {});
+// $ExpectType Promise<Subscribe<Log>>
+eth.subscribe('logs', {}, (error: Error, result: Subscribe<Log>) => {});
+
+// $ExpectType Promise<Subscribe<any>>
+eth.subscribe('syncing');
+// $ExpectType Promise<Subscribe<any>>
+eth.subscribe('syncing', (error: Error, result: Subscribe<any>) => {});
+
+// $ExpectType Promise<Subscribe<BlockHeader>>
+eth.subscribe('newBlockHeaders');
+// $ExpectType Promise<Subscribe<BlockHeader>>
+eth.subscribe('newBlockHeaders', (error: Error, result: Subscribe<BlockHeader>) => {});
+
+// $ExpectType Promise<Subscribe<BlockHeader>>
+eth.subscribe('pendingTransactions');
+// $ExpectType Promise<Subscribe<BlockHeader>>
+eth.subscribe('pendingTransactions', (error: Error, result: Subscribe<Transaction>) => {});
+
+// $ExpectType Providers
+eth.providers;
+
+// $ExpectType AbstractProviderAdapter | provider | null
+eth.givenProvider;
+
+// $ExpectType BatchRequest
+eth.BatchRequest();
```

Comment:
Have you defined the BatchRequest typings in web3-providers?

Reply:
yep 

```
export class BatchRequest {
    constructor(provider: provider, jsonRpcMapper: JsonRpcMapper, jsonRpcResponseValidator: JsonRpcResponseValidator);
    add(request: any): void;
    execute(): void;
    hasOutputFormatter(request: any): boolean;
}
```